src/main/java/org/jbake/template/DelegatingTemplateEngine.java

```diff
@@ -30,8 +32,21 @@
     public DelegatingTemplateEngine(final CompositeConfiguration config, final ContentStore db, final File destination, final File templatesPath) {
         super(config, db, destination, templatesPath);
         this.renderers = new TemplateEngines(config, db, destination, templatesPath);
+        this.registerExtractorsForCustomTypes();
```

Comment:
There was no special reason to add this in the DelegatingTemplateEngine.
It was just convenient. 
The Renderer is the class which instantiates the DelegatingTemplateEngine and is the first AbstractTemplateEngine which instantiates a ModelExtractors class filling up the shared Datastructure holding the references to the different extractors. At this time no new DocumentTypes get registered. That's it.
But you are right, maybe it is better to register the extractors after the DocumentTypes are collected, while warming up the Oven.
It would be more consistent to the Oven analogy.

* heat up
* bake
* cool down